src/client/components/ErrorBoundary/ErrorBoundary.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+class ErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { hasError: false };
+    }
+
+    static getDerivedStateFromError(error) {
+        // Update state so the next render will show the fallback UI.
+        return { hasError: true };
```

Comment:
Here with each component update the state will get hasError, it should be refactored to work as expected